tests/functional/locators.py

```diff
@@ -14,7 +14,7 @@ class NavigationLocators:
 
 class HeaderLocators:
     SEARCH_INPUT = (By.ID, "site-search-text")
-    SEARCH_SUBMIT = (By.CSS_SELECTOR, "header li.search form input.submit")
+    SEARCH_SUBMIT = (By.CSS_SELECTOR, "header form#search-form input.submit")
```

Comment:
@samuelhwilliams we should ideally avoid using id-based selectors (`#`). Could you switch to a class instead?

Reply:
Can you remind me why, please?

Reply:
(And is this really significant enough to block merging?)

Reply:
@samuelhwilliams id selectors have really high ['specificity'](https://drafts.csswg.org/selectors-3/#specificity), so `#a .b` always overrides `.b .c` which can be confusing and sometimes leads to "specificity wars" whereby more and more specific selectors get used to override existing ones. If you keep all css selectors to just use class names, you can avoid this more easily. It's also possibly a cleaner separation of concerns (as IDs are also used as the target of fragment/anchor links, so it can become unclear whether the ID was added for that purpose, or as a CSS styling hook, or for javascript).

That said, we've used `#` in loads of places, but I think we should aim to move more towards the [GOVUK Frontend coding standards](https://github.com/alphagov/govuk-frontend/blob/master/docs/contributing/coding-standards/css.md)?

Reply:
Yeah ok, I understand specificity so I get that we often don't want to use them to apply styling rules. But when we're just looking for an element on a page, it doesn't really seem that bad a thing to do? It lets you grab a very specific element - which in this case, is exactly what we want?

For styling I totally agree and am on board.

Reply:
Although maybe my argument of using IDs to find elements on the page isn't great, particularly if we don't use the ID elsewhere, because then it's added just to test the code, and I guess it could lead to the confusion over what its purpose is, as you described. 🤔